ci.nix

```diff
@@ -0,0 +1,28 @@
+# Should work with any nixpkgs revision, so we don't care
+{ pkgs ? import <nixpkgs> { } }:
+
+let
+  inherit (pkgs) lib haskell;
+  inherit (haskell.lib.compose) buildFromSdist overrideCabal;
+
+  # All non-binary GHC package sets
+  eligible = lib.filterAttrs
+    (name: _:
+      !(lib.hasInfix "Binary" name)
+      && !(lib.hasPrefix "ghcjs" name)
+      && lib.hasPrefix "ghc" name)
+    haskell.packages;
+in
+
+lib.mapAttrs
+  (_: hpkgs:
+    # cabal-install is not cached for all package sets, so we'll rely on Setup.hs
+    lib.pipe
+      (hpkgs.callCabal2nix "jailbreak-cabal-ghc-${hpkgs.ghc.version}.nix" ./. {})
```

Comment:
Why does this work? I thought the name passed to `callCabal2nix` needs to match the name of the `.cabal` file. Is that wrong?

Reply:
That was fixed: https://github.com/NixOS/nixpkgs/commit/5d53087f9b96919eb2175bf44d5669575e24e2f6

Reply:
TIL, thx.

Reply:
It *is* kinda stupid that `name` now has no fixed meaning nor content, but eh.